File: Decorator/Java/DataServiceNoDecorator/src/DataServiceNoDecorator/DataService.java
package DataServiceNoDecorator;

import java.util.logging.Logger;

public class DataService implements IDataService
{
    private final Logger _logger;

    public DataService(Logger logger)
    {
        _logger = logger;
    }
    public String getText()
    {
        _logger.info("Start fetching text");
        var text = "Lorem ipsum";
        _logger.info("End fetching text");

        return text;
    }
}

File: Decorator/C#/DataServiceNoDecorator/DataService.cs
using DataServiceNoDecorator.Infrastructure;

namespace DataServiceNoDecorator
{
    //Dataservice with logging
    public interface IDataService
    {
        string GetText();
    }

    public class DataService : IDataService
    {
        private readonly Logger _logger;

        public DataService(Logger logger)
        {
            _logger = logger;
        }
        public string GetText()
        {
            _logger.Log("Start fetching text");
            var text = "Lorem ipsum";
            _logger.Log("End fetching text");

            return text;
        }
    }
}
